### PR TITLE
Add generic BSON store to Arctic

### PR DESCRIPTION
**Pull request by achamayou**
As discussed before, this is a generic BSON store for Arctic, which is really just exposing the underlying pymongo functionality within the framework of Arctic.

### CODE REVIEW

**Comment by achamayou** [comment]
Has anyone seen this before?

`tests/unit/test_arctic.py::test_arctic_lazy_init 2017-05-30 13:53:26,856 INFO pytest_server_fixtures.mongo Connecting to Mongo at 127.174.134.67:20117
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated`

File: arctic/store/bson_store.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from .._util import enable_sharding
+
+logger = logging.getLogger(__name__)
+
+BSON_STORE_TYPE = 'BSONStore'
+
+class BSONStore(object):
+    """
+    BSON Data Store. This stores any Python object that encodes to BSON correctly,
+    and offers a vanilla pymongo interface. Note that strings myst be valid UTF-8.
```
Comment (AdrianTeng) [inline arctic/store/bson_store.py]: `s/myst/must`

**Comment by AdrianTeng** [comment]
@achamayou Got this on my branch as well -- i wonder if it is travis being jumpy, since this never happens to circle ci

**Comment by AdrianTeng** [comment]
Oh, also, `CHANGES.md` :)

**Pull request merged**